src/cljs/nr/utils.cljs

```diff
@@ -91,3 +92,137 @@
    "Snapshot Plus" "snapshot-plus"
    "SOCR8" "socr8"
    "Casual" "casual"})
+
+(defn map-if [condition f s]
+  "Map a function over elements of a sequence for which condition is true"
+  (map #(if (condition %) (f %) %) s))
+
+(defn regex-escape [string]
+  "Escape characters in a string which have special meanings in regexes"
+  (let [special-chars ".*+?[](){}^$"
+        escaped-chars (map #(str "\\" %) special-chars)
+        regex-escape-smap (zipmap special-chars escaped-chars)]
+    (->> string
+         (replace regex-escape-smap)
+         (reduce str))))
+
+(def icon-smap
+  "A map of case insensitive icon regexes to icon span fragments"
+  (letfn [(span-of [icon] [:span {:class (str "anr-icon " icon)}])
+          (regex-of [icon-code] (re-pattern (str "(?i)" (regex-escape icon-code))))]
+    (->> {"[credit]" "credit"
+          "[credits]" "credit"
+          "[c]" "credit"
+          "[recurring credit]" "recurring-credit"
+          "[recurring credits]" "recurring-credit"
+          "[recurring-credit]" "recurring-credit"
+          "[recurring-credits]" "recurring-credit"
+          "[click]" "click"
+          "[clicks]" "click"
+          "1[memory unit]" "mu1"
+          "1[mu]" "mu1"
+          "2[memory unit]" "mu2"
+          "2[mu]" "mu2"
+          "3[memory unit]" "mu3"
+          "3[mu]" "mu3"
+          "[memory unit]" "mu"
+          "[mu]" "mu"
+          "[link]" "link"
+          "[l]" "link"
+          "[subroutine]" "subroutine"
+          "[trash]" "trash"
+          "[t]" "trash"
+          "[adam]" "adam"
+          "[anarch]" "anarch"
+          "[apex]" "apex"
+          "[criminal]" "criminal"
+          "[hb]" "haas-bioroid"
+          "[haas-bioroid]" "haas-bioroid"
+          "[jinteki]" "jinteki"
+          "[nbn]" "nbn"
+          "[shaper]" "shaper"
+          "[sunny]" "sunny"
+          "[weyland]" "weyland-consortium"
+          "[weyland-consortium]" "weyland-consortium"}
+      (map (fn [[k v]] [(regex-of k) (span-of v)])))))
+
+(defn card-smap-impl []
+  "A map of case sensitive card regexes to card span fragments"
+  (letfn [(unpack-codes [card] [:title card :code card])
```

Comment:
Watch out for lack of `()`, as this will evaluate to `[:title {:title "Card Name" ...} :code {:title "Card Name" ...}]`

Reply:
Now using `juxt` as you suggested in slack :+1: